src/filters/factory.rs

```diff
@@ -24,6 +24,23 @@ use crate::{
 /// An owned pointer to a dynamic [`FilterFactory`] instance.
 pub type DynFilterFactory = Box<dyn FilterFactory>;
 
+/// The value returned by [`FilterFactory::create_filter`].
+pub struct CreatedFilter {
+    /// The configuration used to create the filter.
+    pub config: serde_json::Value,
```

Comment:
Since most of the time we create a new a filter the configuration doesn't change, as a future improvement, we could use `Arc` here instead to share memory across filter instances.